features/src/autogluon/features/generators/drop_duplicates.py

```diff
@@ -12,9 +13,35 @@
 
 # TODO: Not necessary to exist after fitting, can just update outer context feature_out/feature_in and then delete this
 class DropDuplicatesFeatureGenerator(AbstractFeatureGenerator):
-    """Drops features which are exact duplicates of other features, leaving only one instance of the data."""
+    """
+    Drops features which are exact duplicates of other features, leaving only one instance of the data.
+
+    Parameters
+    ----------
+    sample_size_init : int, default 1000
+        The number of rows to sample when doing an initial filter of duplicate feature candidates.
+        Usually, the majority of features can be filtered out using this smaller amount of rows which greatly speeds up the computation of the final check.
+        If None or greater than the number of rows, no initial filter will occur. This may increase the time to fit immensely for large datasets.
+    sample_size_final : int, default 20000
```

Comment:
This could probably be even lower (say 5000), but I made it 20K to ensure that only under the most absurd situations would it incorrectly detect duplicates. We can easily drop it to 5000 if users find it to still be too slow in certain situations.

Reply:
Great idea, I had something similar in mind, but this is better.